jlm/llvm/opt/inlining.cpp

```diff
@@ -142,22 +142,21 @@ inlineCall(jlm::rvsdg::simple_node * call, const lambda::node * lambda)
 }
 
 static void
-inlining(jlm::rvsdg::graph & graph)
+inlining(jlm::rvsdg::graph & rvsdg)
 {
-	auto root = graph.root();
-
-	for (auto node : jlm::rvsdg::topdown_traverser(root)) {
-		if (!is<lambda::operation>(node))
-			continue;
-
-		auto lambda = static_cast<const lambda::node*>(node);
-
-		std::vector<jlm::rvsdg::simple_node*> calls;
-		bool onlyDirectCalls = lambda->direct_calls(&calls);
-
-		if (onlyDirectCalls && calls.size() == 1)
-			inlineCall(calls[0], lambda);
-	}
+  for (auto node : rvsdg::topdown_traverser(rvsdg.root()))
+  {
+    if (auto lambda = dynamic_cast<const lambda::node*>(node))
+    {
+      auto callSummary = lambda->ComputeCallSummary();
```

Comment:
this is already traversing the graph, and "ComputeCallSummary" performs some inner traversal. I guess it "could" become quadratic for pathological cases of deeply nested lambdas, but probably it is not worth doing anything about this.

Reply:
Good point, but I also agree with your assessment that it is not worth doing something about it (at least so far).